### ISSUE TITLE
Shorting the returned code in the Authorisation Code with PKCE implementation

### ISSUE THREAD

**Issue by AKlaus** [AKlaus/Transparent-Auth-Gateway#1]
**CONTEXT**: the _Authorisation Code with PKCE_ ([RFC 7636](https://www.rfc-editor.org/rfc/rfc7636)) implementation with no authorization storage (e.g. for the _Degraded mode_) and hence no reference tokens in use.

**PROBLEM**
The server returns the _Code_ (encrypted by default) wrapped in a signed JWT by calling [link to GenerateIdentityModelToken.ValueTask()](https://github.com/openiddict/openiddict-core/blob/7915f28d90d61fa7a3d34ed20ca8b56c0dd5eb26/src/OpenIddict.Server/OpenIddictServerHandlers.Protection.cs#L1296) method: 
`context.Token = context.SecurityTokenHandler.CreateToken(descriptor)`.
This (especially the signature) blows out the size of the generated _code_ string (becomes > 2,048 symbols) that gets returned in the _Query String_.

**TL;DR**: Due to the heavy size of the token, in its turn, the client's request may hit limitations on the _Query String_ and/or _URL_ length.

**GOAL**: 
Keep the generated _code_ string well under 2,048 symbols.
e.g. `code=lBm8NqlNa0qKalXTRc7cYA&state=V2VkIEp1bCAyNiAyMDIzIDE4OjE3OjMzIEdNVCsxMDAwIChBdXN0cmFsaWFuIEVhc3Rlcm4gU3RhbmRhcmQgVGltZSk%3D&iss=https%3A%2F%2Flocalhost%3A5003%2F`

**Comment by AKlaus** [comment]
Cross-link: openiddict/openiddict-core/issues/1843
